lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2445,6 +2445,22 @@ class DecomposeAten_EmbeddingBagOp
 };
 } // namespace
 
+namespace {
+class DecomposeAtenMaskedFillTensorOp
+    : public OpRewritePattern<AtenMaskedFillTensorOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMaskedFillTensorOp op,
+                                PatternRewriter &rewriter) const override {
+    Value ScalarFloat =
+        rewriter.create<AtenFloatTensorOp>(op.getLoc(), op.value());
```

Comment:
This only works if the fill value is a float, which is not guaranteed.

I think it would make more sense to add support for this op similar to https://github.com/llvm/torch-mlir/blob/072c2b5aafdd5858bb19c5b296453a9cf0b65113/lib/Conversion/TorchToLinalg/Uncategorized.cpp#L867. This would require the same, if not less, amount of code, and it would have the added benefit that it would work for multiple dtypes for the fill value.

Reply:
I see, this makes sense. Thanks.